src/meshapi/util/join_records.py

```diff
@@ -98,3 +120,46 @@ def flush_test_data(self) -> None:
             print(f"Folder '{folder_path}' deleted from bucket '{JOIN_RECORD_BUCKET_NAME}'.")
         else:
             print(f"No objects found in folder '{folder_path}'.")
+
+    # Examines join records from pre- and post-submission and ensures a copy is
+    # present in each. If one is missing from the post-submission, indicating there
+    # was some deeper problem in submitting to MeshDB, we backfill it into
+    # the post-submission list we return
+    def ensure_pre_post_consistency(self, since: datetime.datetime) -> dict[str, JoinRecord]:
+        # Get join records from both pre and post submissions so we can make sure
+        # there are no discrepancies
+
+        # Grab a copy of pre, post, and replayed submissions
+        # Convert the lists to dictionaries to make them easier to search by UUID
+        pre_join_records_dict: dict[str, JoinRecord] = {}
+        for record in self.get_all(since, SubmissionStage.PRE):
+            if record.version >= 2:
+                pre_join_records_dict[record.uuid] = record
+
+        post_join_records_dict: dict[str, JoinRecord] = {}
+        for record in self.get_all(since, SubmissionStage.POST):
+            if record.version >= 2:
+                post_join_records_dict[record.uuid] = record
+
+        # For each pre-submission record, there should exist a post-submission
+        # record. If the post-submission record is missing, log a warning and add
+        # the pre-submission record to the post-submission dictionary so it is covered
+        # by the command.
```

Comment:
If we're checking consistency, would it be good to check that every post record has a pre record also? Not sure how we should handle that but it could theoretically happen it we probably should at least have a log statement